go/stellar/relay.go

```diff
@@ -0,0 +1,123 @@
+package stellar
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/protocol/stellar1"
+	"github.com/keybase/client/go/stellar/remote"
+	"github.com/keybase/client/go/teams"
+	"github.com/keybase/stellarnet"
+	"github.com/stellar/go/build"
+	"github.com/stellar/go/keypair"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+// Get the key used to encrypted the key for a relay transfer
+// A key from the implicit team betwen the logged-in user and `to`.
+// If `generation` is nil, gets the latest key.
+// TODO CORE-7718 make this private
+func KeyForRelayTransfer(ctx context.Context, g *libkb.GlobalContext, remoter remote.Remoter,
+	other *libkb.User, generation *keybase1.PerTeamKeyGeneration) (res keybase1.TeamApplicationKey, err error) {
+	if other == nil {
+		return res, fmt.Errorf("missing other user")
+	}
+	meUsername, err := g.GetUPAKLoader().LookupUsername(ctx, g.ActiveDevice.UID())
+	if err != nil {
+		return res, err
+	}
+	impTeamDisplayName := fmt.Sprintf("%s,%s", meUsername.String(), other.GetNormalizedName().String())
```

Comment:
Do we have something elsewhere to make names like these?

Reply:
We have a struct `keybase1.ImplicitTeamDisplayName` which could make sense.
Then could this func [LookupOrCreateImplicitTeam here](https://github.com/keybase/client/blob/master/go/teams/implicit.go#L176). But I wouldn't know what to call the func `LookupOrCreateImplicitTeamFromStructName`